### DISCUSSION TITLE
Make postUpgradeTasks fileFilters more user friendly when working with dotfiles

### DISCUSSION THREAD

**Discussion by bgutschke** [renovatebot/renovate#21220] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

Hello everyone,

I would like to propose a change on how postUpgradeTasks fileFilters work.

**Problem brief**:
I am using Renovate in combination with custom [Nx migrations](https://nx.dev/packages/nx/documents/migrate). Basically having an npm package which is executed through postUpgradeTasks, making changes to the repository/file structure.

From time to time I have to add/update/delete dotfiles in those migrations/postUpgradeTasks.

At the moment configuring renovate to not ignore those dotfiles is rather cumbersome and error prone.

Given the following file structure:
```
- .ops/test.yml
- .ops/includes/.gitlab.yml
- apps/test-app/.eslintrc
- apps/test-app/project.json
- .gitlabci.yml
```
the fileFilter would have to look something like this:
```json
fileFilter: [
  "**/**",
  "**/*."
  ".*",
  ".*/**",
  ".*/**/.*"
]
```

This can get even more complicated depending on how dot-directories and -files are nested.

This is due to how [minimatch](https://github.com/isaacs/minimatch) treats dotfiles: https://github.com/isaacs/minimatch#dot
Imho this is pretty unintuitive and not very readable.

**Possible solutions I could think of**:
- Use minimatch with enabled `dot` option [here](https://github.com/renovatebot/renovate/blob/d869c946d164965393f1259db5171e2a0303fb27/lib/workers/repository/update/branch/execute-post-upgrade-commands.ts#L109) and [here](https://github.com/renovatebot/renovate/blob/d869c946d164965393f1259db5171e2a0303fb27/lib/workers/repository/update/branch/execute-post-upgrade-commands.ts#L137) `minimatch(relativePath, pattern, { dot: true })`
- Expose the (optional?) dot configuration of minimatch through the renovate configuration

**Last thing**:
The documentation on [`fileFilters`](https://docs.renovatebot.com/configuration-options/#filefilters) is pretty minimal. Maybe it could use a paragraph about dotfiles?

Thanks for reading this if you made it this far and thanks for Renovate.

Bastian

**Comment by viceice** [comment]
I think enabling `dot` option will be ok, but it's probably a breaking change 🤔

**Reply by bgutschke** [reply]
That would be great. And yes, I would agree that this would be a breaking change.

**Comment by rarkins** [comment]
Created feature request issue: #21276
